File: my-folder/0205-isomorphic-strings/solution.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.size() != t.size()) return false;
        
        int m1[256] = {0}, m2[256] = {0};
        int count = 1;
        
        for(int i = 0; i < s.size(); i++){
            if(m1[s[i]] != m2[t[i]]) return false;
            
            if(m1[s[i]] == 0){
                m1[s[i]] = count;
                m2[t[i]] = count;
                count++;
            }
        }
        
        return true;
    }
};

File: my-folder/0205-isomorphic-strings/solution.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character,Character> mp = new HashMap<>();
        int i = 0,j = 0;
        while(i<s.length()){
            char ct = s.charAt(i);
            char st = t.charAt(i);
            if(!mp.containsKey(ct) && !mp.containsValue(st)){
                mp.put(ct,st);
            }
            else if(mp.containsValue(st) && !mp.containsKey(ct)){
                return false;
                
            }
            else if(mp.containsKey(ct)){
                if(mp.get(ct)!=st) return false;
            }
            i++;
        }
        return true;
        
    }
}
